lib/modules/manager/dockerfile/extract.ts

```diff
@@ -162,69 +233,160 @@ export function getDep(
 export function extractPackageFile(content: string): PackageFile | null {
   const deps: PackageDependency[] = [];
   const stageNames: string[] = [];
+  const args: Record<string, string> = {};
+  const argsLines: Record<string, number[]> = {};
 
-  const fromMatches = content.matchAll(
-    /^[ \t]*FROM(?:\\\r?\n| |\t|#.*?\r?\n|[ \t]--[a-z]+=\S+?)*[ \t](?<image>\S+)(?:(?:\\\r?\n| |\t|#.*\r?\n)+as[ \t]+(?<name>\S+))?/gim // TODO #12875 complex for re2 has too many not supported groups
-  );
+  let escapeChar = '\\\\';
+  let lookForEscapeChar = true;
+
+  const lineFeed = content.indexOf('\r\n') >= 0 ? '\r\n' : '\n';
+  const lines = content.split(newlineRegex);
+  for (let lineNumber = 0; lineNumber < lines.length; ) {
+    const lineNumberInstrStart = lineNumber;
+    let instruction = lines[lineNumber];
 
-  for (const fromMatch of fromMatches) {
-    if (fromMatch.groups?.name) {
-      logger.debug('Found a multistage build stage name');
-      stageNames.push(fromMatch.groups.name);
+    if (lookForEscapeChar) {
+      const directivesMatch = regEx(
+        /^[ \t]*#[ \t]*(?<directive>syntax|escape)[ \t]*=[ \t]*(?<escapeChar>\S)/i
```

Comment:
maybe its cleaner if we move this regex to a variable and give it name for what it does

Reply:
You mean the expression `/^[ \t]*#[ \t]*(?<directive>syntax...` itself? To be honest, I'd suggest to keep it here for consistency with the other RegExes. `directivesMatch` verbally refers to Dockerfile [parser directives](https://docs.docker.com/engine/reference/builder/#parser-directives), so the naming should already be very descriptive. Still, please let me know, if it's imperative for you to have this changed.

Reply:
i see your point now worries not necessary